components/datetime/src/format/datetime.rs

```diff
@@ -215,197 +241,375 @@ const PLACEHOLDER_LEAP_PREFIX: &str = "(leap)";
+                ),
                 field.length,
-            )?,
-            Week::WeekOfMonth => format_number(
+            )?
+            .unwrap_or_else(|e| {
+                if r.is_ok() {
+                    r = Err(e);
+                }
+            }),
+            Week::WeekOfMonth => try_write_number(
                 w,
                 fixed_decimal_format,
-                FixedDecimal::from(datetime.week_of_month()?.0),
+                FixedDecimal::from(
+                    datetime
+                        .week_of_month()
+                        .unwrap_or_else(|e| {
+                            if r.is_ok() {
+                                r = Err(Error::DateTimeInput(e));
+                            }
+                            todo!("fallback value")
+                        })
+                        .0,
+                ),
                 field.length,
-            )?,
+            )?
+            .unwrap_or_else(|e| {
+                if r.is_ok() {
+                    r = Err(e);
+                }
+            }),
         },
         FieldSymbol::Weekday(weekday) => {
-            let dow = datetime
-                .datetime()
-                .iso_weekday()
-                .ok_or(Error::MissingInputField(Some("iso_weekday")))?;
+            let dow = datetime.datetime().iso_weekday().unwrap_or_else(|| {
+                if r.is_ok() {
+                    r = Err(Error::MissingInputField(Some("iso_weekday")));
+                }
+                todo!("fallback value")
+            });
             let symbol = date_symbols
-                .ok_or(Error::MissingDateSymbols)?
-                .get_symbol_for_weekday(weekday, field.length, dow)?;
+                .unwrap_or_else(|| {
+                    if r.is_ok() {
+                        r = Err(Error::MissingDateSymbols);
+                    }
+                    todo!("fallback value")
+                })
+                .get_symbol_for_weekday(weekday, field.length, dow)
+                .unwrap_or_else(|e| {
+                    if r.is_ok() {
+                        r = Err(e);
+                    }
+                    todo!("fallback value")
+                });
             w.write_str(symbol)?
         }
-        symbol @ FieldSymbol::Day(day) => format_number(
+        symbol @ FieldSymbol::Day(day) => try_write_number(
             w,
             fixed_decimal_format,
             FixedDecimal::from(match day {
                 fields::Day::DayOfMonth => {
                     datetime
                         .datetime()
                         .day_of_month()
-                        .ok_or(Error::MissingInputField(Some("day_of_month")))?
+                        .unwrap_or_else(|| {
+                            if r.is_ok() {
+                                r = Err(Error::MissingInputField(Some("day_of_month")));
+                            }
+                            todo!("fallback value")
+                        })
```

Comment:
Suggestion: Currently the code calls `try_write_number` with this inner argument that must resolve to a number. I think we may want to change the nesting so that we resolve the inner argument first and then call `try_write_number` at the end. This makes it more clear how we can continue to the next iteration of the loop if there was a missing input field.

Reply:
done